lib/actions/write.js

```diff
@@ -0,0 +1,127 @@
+const request = require('request-promise');
+const co = require('co');
+const csv = require('csv');
+const _ = require('lodash');
+const messages = require('elasticio-node').messages;
+const client = require('elasticio-rest-node')();
+
+let stringifier;
+let signedUrl;
+let timeout;
+let rowCount = 0;
+
+exports.init = init;
+exports.process = processAction;
+
+function init(cfg) {
+    return co(function* gen() {
+
+        const delimiter = cfg.reader.delimiter || ',';
+        console.log('Using delimiter: \'%s\'', delimiter);
+        const options = {
+            header: true,
+            delimiter
+        };
+
+        if (cfg.reader.columns) {
+
+            const columns = Object.keys(_.keyBy(cfg.reader.columns, 'property'));
+
+            console.log('Configured column names:', columns);
+            options.columns = columns;
+        }
+
+        stringifier = csv.stringify(options);
+
+        signedUrl = yield client.resources.storage.createSignedUrl();
+
+        let putUrl = signedUrl.put_url;
+        console.log('CSV file to be uploaded file to uri=%s', putUrl);
+
+        stringifier.pipe(request.put(putUrl));
+    });
+}
+
+function processAction(msg, cfg) {
+    const self = this;
+
+    const startRow = cfg.reader.startRow || 0;
+
+    if (startRow !== 0 && rowCount === 0) {
```

Comment:
Current implementation of `startRow` is confusing.
It's better to either remove it or rewrite.

Reply:
Cann @drobiazko or @zubairov confirm that the startRow setting can be removed?

Reply:
Can we open an enhancement request concerning the ``startRow`` to discuss what's wrong with it? Based on the results of that discussion it will be either removed or kept.